File: solutions/csharp/square-root/1/SquareRoot.cs
public static class SquareRoot
{
    public static int Root(int number)
    {
        int rootMin = 1;
        int rootMax = number;
        while(rootMax - rootMin > 1){
            int tmp = (rootMin + rootMax) / 2;
            if (tmp * tmp <= number){
                rootMin = tmp;
            }
            else{
                rootMax = tmp;
            }
        }
        return rootMin;
    }
}

File: solutions/c/square-root/1/square_root.c
#include "square_root.h"

int square_root(int number){
    int rootUpperBound = 0;
    while(rootUpperBound * rootUpperBound < number){
        ++rootUpperBound;
    }
    if(rootUpperBound * rootUpperBound == number)return rootUpperBound;
    else return 0;
    //Some code to handle the operaions
}
